x/protocolpool/keeper/genesis.go

```diff
@@ -77,6 +77,21 @@ func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) error
 }
 
 func (k Keeper) ExportGenesis(ctx context.Context) (*types.GenesisState, error) {
+	// refresh all funds
+	if err := k.IterateAndUpdateFundsDistribution(ctx); err != nil {
```

Comment:
Let's add a test case for this new logic.

In fact, we don't have a test at all for ExportGenesis.

Reply:
Yep yep, that was the plan